opengrok-web/src/test/java/org/opengrok/web/api/v1/controller/ConfigurationControllerTest.java

```diff
@@ -236,4 +241,50 @@ public void testSuggesterServiceNotifiedOnConfigurationChange() {
         verify(suggesterService).refresh();
     }
 
+    @Test
+    public void testConfigValueSetVsMultiThread() throws InterruptedException {
+        int origValue = env.getHitsPerPage();
+        int nThreads = 128;
+        int values[] = new int[nThreads];
+
+        // Create a thread pool.
+        final CountDownLatch startLatch = new CountDownLatch(nThreads);
+        ExecutorService executor = Executors.newFixedThreadPool(nThreads);
+        for (int i = 0; i < nThreads; i++) {
+            final int ii = i;
+            executor.submit(new Runnable() {
+                @Override
+                public void run() {
+                    // Wait for hint for termination, save the value and exit.
+                    try {
+                        startLatch.await();
+                    } catch (InterruptedException e) {
+                        e.printStackTrace();
+                    }
+                    values[ii] = env.getHitsPerPage();
+                }
+            });
+        }
+
+        // Set a configuration variable.
+        int newValue = 42;
+        setValue("hitsPerPage", String.valueOf(newValue));
+
+        // Unblock the threads.
+        for (int i = 0; i < nThreads; i++) {
+            startLatch.countDown();
+        }
+
+        // Terminate the threads
+        executor.awaitTermination(10, TimeUnit.SECONDS);
+        executor.shutdown();
+
+        // Check thread's view of the variable.
+        for (int j = 0; j < values.length; j++) {
+            assertEquals(newValue, values[j]);
+        }
+
+        // Revert back to default.
+        env.setHitsPerPage(origValue);
+    }
```

Comment:
If you try it without the fix, does this test usually fail or?

Reply:
I think it would need to call `register()` so that each thread has its own distinct configuration snapshot.

Reply:
Using `PageConfig` should do the trick, let me try.

Reply:
(it fails if you remove the hotfix for it)

Reply:
changed the test case so that it is simpler and actually fails without the fix in place.